app/src/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -267,34 +267,42 @@ public static boolean scaleImage(File originalImage, String finalFilePath, int m
         return scaledImage;
     }
 
-
-
     public static Bitmap getBitmapScaledToDisplay(File f, int screenHeight, int screenWidth) {
-        // Determine image size of f
+        // Determine dimensions of original image
         BitmapFactory.Options o = new BitmapFactory.Options();
         o.inJustDecodeBounds = true;
         BitmapFactory.decodeFile(f.getAbsolutePath(), o);
+        int imageHeight = o.outHeight;
+        int imageWidth = o.outWidth;
 
-        int heightScale = o.outHeight / screenHeight;
-        int widthScale = o.outWidth / screenWidth;
-
-        // Powers of 2 work faster, sometimes, according to the doc.
-        // We're just doing closest size that still fills the screen.
+        // Get a scale-down factor -- Powers of 2 work faster according to the docs, but we're
+        // just doing closest size that still fills the screen
+        int heightScale = Math.round((float)imageHeight / screenHeight);
+        int widthScale = Math.round((float)imageWidth / screenWidth);
         int scale = Math.max(widthScale, heightScale);
+        if (scale == 0) {
+            // Rounding could possibly have resulted in a scale factor of 0, which is invalid
+            scale = 1;
+        }
+
+        return performSafeScaleDown(f, scale);
+    }
 
-        // get bitmap with scale ( < 1 is the same as 1)
+    /**
+     * Returns a scaled-down bitmap for the given image file, progressively increasing the
+     * scale-down factor by 1 until allocating memory for the bitmap does not cause an OOM error
+     */
+    private static Bitmap performSafeScaleDown(File f, int scale) {
         BitmapFactory.Options options = new BitmapFactory.Options();
         options.inSampleSize = scale;
-        Bitmap b = BitmapFactory.decodeFile(f.getAbsolutePath(), options);
-        if (b != null) {
-        Log.i(t,
-            "Screen is " + screenHeight + "x" + screenWidth + ".  Image has been scaled down by "
-                    + scale + " to " + b.getHeight() + "x" + b.getWidth());
+        try {
+            return BitmapFactory.decodeFile(f.getAbsolutePath(), options);
+        } catch (OutOfMemoryError e) {
+            scale++;
+            return performSafeScaleDown(f, scale);
```

Comment:
How many times do you suspect it will need to recursively call itself?  Is there any way that this could loop forever? Would it be worth aborting after some number of attempts or a given scale size?